irohad/ametsuchi/impl/wsv_restorer_impl.cpp

```diff
@@ -110,55 +110,38 @@ namespace {
       iroha::validation::ChainValidator &validator,
       HeightType starting_height,
       HeightType ending_height) {
-    auto blocks = rxcpp::observable<>::create<
-        std::shared_ptr<shared_model::interface::Block>>([&block_query,
-                                                          &interface_validator,
-                                                          &proto_validator,
-                                                          starting_height,
-                                                          ending_height](
-                                                             auto s) {
-      for (auto height = starting_height; height <= ending_height; ++height) {
-        auto result = block_query.getBlock(height);
-        if (auto e = iroha::expected::resultToOptionalError(result)) {
-          s.on_error(std::make_exception_ptr(
-              std::runtime_error(std::move(e).value().message)));
-          return;
-        }
+    for (auto height = starting_height; height <= ending_height; ++height) {
+      auto result = block_query.getBlock(height);
+      if (auto e = iroha::expected::resultToOptionalError(result)) {
+        return iroha::expected::makeError(std::move(e).value().message);
+      }
```

Comment:
Жахіття. Це має виглядати якось так
```suggestion
      if (result.isError()) {
        return result;
      }
```